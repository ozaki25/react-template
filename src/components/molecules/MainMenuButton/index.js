import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

function MainMenuButton({ title, description, onClick }) {
  return (
    <Button onClick={onClick}>
      <Title>{title}</Title>
      <Description>{description}</Description>
    </Button>
  );
}

export const Button = styled.button`
  border: solid 1px #eee;
  box-shadow: 2px 2px 2px #bdbdbd;
  width: 300px;
  display: inline-block;
  &:active {
    opacity: 0.5;
    box-shadow: 2px 2px 2px #f5f5f5;
  }
`;

export const Title = styled.p`
  background-color: #eee;
  color: #212121;
  font-size: 14px;
  font-weight: bold;
  padding: 10px;
  text-align: left;
`;

export const Description = styled.p`
  color: #212121;
  font-size: 12px;
  padding: 10px;
  text-align: left;
`;

MainMenuButton.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MainMenuButton;
