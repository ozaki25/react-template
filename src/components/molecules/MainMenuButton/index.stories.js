import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';

import MainMenuButton from '.';

const stories = storiesOf('molecules/MainMenuButton', module);

stories.add(
  '通常パターン',
  withInfo('メインの導線を開始するメニューボタン')(() => (
    <MainMenuButton
      title={text('title', '開始される手続きのタイトル')}
      description={text('description', '手続きの概要をここに書きます')}
      onClick={action('click button')}
    />
  )),
);
