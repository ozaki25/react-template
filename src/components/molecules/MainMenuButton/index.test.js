import React from 'react';
import { snapshot } from 'test/helpers';
import MainMenuButton, { Button, Title, Description } from '.';

const title = 'タイトル';
const description = '説明';
const onClick = jest.fn();

const props = {
  title,
  description,
  onClick,
};

snapshot('StartReissueButton', <MainMenuButton {...props} />);
snapshot('StartReissueButton/Button', <Button onClick={onClick} />);
snapshot('StartReissueButton/Title', <Title>{title}</Title>);
snapshot('StartReissueButton/Description', <Description>{description}</Description>);
