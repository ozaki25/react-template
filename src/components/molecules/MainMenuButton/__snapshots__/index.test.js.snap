// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: StartReissueButton 1`] = `
<styled.button
  onClick={[Function]}
>
  <styled.p>
    タイトル
  </styled.p>
  <styled.p>
    説明
  </styled.p>
</styled.button>
`;

exports[`should match snapshot: StartReissueButton/Button 1`] = `
.c0 {
  border: solid 1px #eee;
  box-shadow: 2px 2px 2px #bdbdbd;
  width: 300px;
  display: inline-block;
}

.c0:active {
  opacity: 0.5;
  box-shadow: 2px 2px 2px #f5f5f5;
}

<button
  className="c0"
  onClick={[Function]}
/>
`;

exports[`should match snapshot: StartReissueButton/Description 1`] = `
.c0 {
  color: #212121;
  font-size: 12px;
  padding: 10px;
  text-align: left;
}

<p
  className="c0"
>
  説明
</p>
`;

exports[`should match snapshot: StartReissueButton/Title 1`] = `
.c0 {
  background-color: #eee;
  color: #212121;
  font-size: 14px;
  font-weight: bold;
  padding: 10px;
  text-align: left;
}

<p
  className="c0"
>
  タイトル
</p>
`;
